HACKING.md

```diff
@@ -354,24 +383,39 @@ git checkout -b <branchname>
 ```
 
 Make changes to the repository and commit the changes to your
-working branch. Push the changes to your forked `snapd` repository.
+local working branch.
+
+```sh
+git commit -sa -m "<feature/subject-prefix>: <brief-description-of-the-change>
+
+<more-elaborate-description-with-reasoning-about-the-changed>"
+``` 
+
+>If it's justified, one might consider several commits before pushing 
+their work to the remote repository. 
+One should strive to have only one commit per topic, and to avoid pushing 
+many topics in the same attempt.  
+
+Push the changes to your forked `snapd` repository.
 
 ```
-git commit -a -m "commit message"
-git push fork <branchname>
+git push fork <local-branchname>[:<remote-branchname>]
 ```
 
+>Note that if the `<remote-branchname>` is not specified, a new remote branch 
+is created implicitly, by the name of the `<local-branchname>`.
+
 Create the Pull Request for your branch on GitHub.
 
 This complete process is outlined in the GitHub documentation [here](
 https://docs.github.com/en/github/collaborating-with-pull-requests).
 
-We value good tests, so when you fix a bug or add a new feature we highly
-encourage you to add tests.
+## Testing
 
-For more information on testing, please see the Testing section.
+### Running unit-tests
```

Comment:
I would move this after the next paragraph.

Reply:
What do you mean? Moving unit-tests after integration tests?

Reply:
Sure, I get it. You're right. The first paragraph of the section should address the whole set of testing, not just unit-testing. My bad. Thank you, for pointing it out.